File: src/main/java/gg/grounds/repo/ProxyHeartbeatRepository.java
package gg.grounds.repo;

import gg.grounds.domain.model.ProxyHeartbeatEntity;
import io.quarkus.hibernate.reactive.panache.PanacheRepositoryBase;
import io.smallrye.mutiny.Uni;
import jakarta.enterprise.context.ApplicationScoped;

import java.time.Instant;

@ApplicationScoped
public class ProxyHeartbeatRepository implements PanacheRepositoryBase<ProxyHeartbeatEntity, String> {

    public Uni<Long> countOffline(Instant cutoff) {
        return count("lastSeenAt < ?1", cutoff);
    }
}

File: src/main/kotlin/gg/grounds/repo/ProxyHeartbeatRepository.kt
package gg.grounds.repo

import gg.grounds.domain.model.ProxyHeartbeatEntity
import io.quarkus.hibernate.reactive.panache.PanacheRepositoryBase
import io.smallrye.mutiny.Uni
import jakarta.enterprise.context.ApplicationScoped
import java.time.Instant

@ApplicationScoped
class ProxyHeartbeatRepository : PanacheRepositoryBase<ProxyHeartbeatEntity?, String?> {
    fun countOffline(cutoff: Instant?): Uni<Long?>? {
        return count("lastSeenAt < ?1", cutoff)
    }
}
